src/workerd/jsg/value.h

```diff
@@ -851,6 +851,67 @@ class ArrayWrapper {
   }
 };
 
+// =======================================================================================
+// Sets
+
+// TypeWrapper mixin for sets.
+template <typename TypeWrapper>
+class SetWrapper {
+ public:
+  static auto constexpr MAX_STACK = 64;
+  template <typename U>
+  static constexpr const char* getName(kj::HashSet<U>*) {
+    return "Set";
+  }
+
+  template <typename U>
+  v8::Local<v8::Value> wrap(v8::Local<v8::Context> context,
+      kj::Maybe<v8::Local<v8::Object>> creator,
+      kj::HashSet<U> set) {
+    v8::Isolate* isolate = context->GetIsolate();
+    v8::EscapableHandleScope handleScope(isolate);
+
+    auto out = v8::Set::New(isolate);
+    for (const auto& item: set) {
+      out = check(out->Add(context,
+          static_cast<TypeWrapper*>(this)->wrap(context, creator, item).template As<v8::Value>()));
+    }
+
+    return handleScope.Escape(out);
+  }
+
+  template <typename U>
+  v8::Local<v8::Value> wrap(v8::Local<v8::Context> context,
+      kj::Maybe<v8::Local<v8::Object>> creator,
+      kj::HashSet<U>& set) {
+    return static_cast<TypeWrapper*>(this)->wrap(context, creator, set.asPtr());
```

Comment:
Does this overload actually work? I don't think `set` has an `asPtr()` method?

Reply:
Hm, maybe not, I'll just remove this overload.